File: str/isomorphicStr1.java
//follow up: https://www.youtube.com/watch?v=MNDOsYgVRyY
//mapping is 'a' <-> 'x', 'b' <-> 'y', 'c' <-> 'z'.
public boolean isomorphic(String s, String t) {
    // Write your solution here
    if(s == null||s.length()<=1)
      return true;
    HashMap<Character, Character> map = new HashMap<>();
    for(int i = 0; i < s.length; i++){
      char a = s.charAt(i);
      char b = t.charAt(i);
      if(map.containsKey(a)){
        if(map.get(a).equals(b))
          continue;
        else
         return false;
      }else{
        if(!map.containsValue(b)){
          map.put(a, b)
        else
          return false;
        }
      }
    }
    return true;
  }

File: str/isomorphicStr1.py
#follow up: https://www.youtube.com/watch?v=MNDOsYgVRyY
# mapping is 'a' <-> 'x', 'b' <-> 'y', 'c' <-> 'z'.
def ismorphic(s, t):
    if s is None or len(s) <= 1:
        return True
    dict = {}
    s_arr = list(s)
    t_arr = list(t)
    for index in len(s_arr):
        if s_arr[index] in dict:
            if dict[s_arr[index]] != t_arr[index]:
                return False
        else:
            if t_arr[index] in dict.values():
                return False
            else:
                dict[s_arr[index]] = t_arr[index]
    return True
